### PR TITLE
[cowprotocol] Fix trades, batches and solver models

### PR DESCRIPTION
**Pull request by fleupold**
### Description:

This PR aligns the different queries we have to compute cow protocol trades, batches and solvers across network so that they can be run with minimal `diff`. This is useful when adding new networks.

Most notable changes are
- ETH flow (buy and sell) orders were only tracked on mainnet
- Removed unnecessary additional solver query for non-mainnet batch table
- Remaining formatting changes are for reducing `diff` further

---
quick links for more information:
- [README.md](https://github.com/duneanalytics/spellbook/blob/main/README.md)
- [spellbook docs](https://github.com/duneanalytics/spellbook/tree/main/docs)
- [CONTRIBUTING.md](https://github.com/duneanalytics/spellbook/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: dbt_subprojects/dex/models/_projects/cow_protocol/arbitrum/cow_protocol_arbitrum_trades.sql [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+    partition_by = ['block_month'],
+    unique_key = ['tx_hash', 'order_uid', 'evt_index'],
+    on_schema_change='sync_all_columns',
+    file_format ='delta',
+    incremental_strategy='merge',
+    incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+    post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                "project",
+                                "cow_protocol",
+                                \'["olgafetisova"]\') }}'
```
Comment (bh2smith) [inline dbt_subprojects/dex/models/_projects/cow_protocol/arbitrum/cow_protocol_arbitrum_trades.sql:14]: Don't forget to include yourself!
```suggestion
                                \'["olgafetisova", "fleupold"]\') }}'
```

File: dbt_subprojects/dex/models/_projects/cow_protocol/gnosis/cow_protocol_gnosis_trades.sql [hunk 2]
```diff
@@ -167,9 +186,10 @@ valued_trades as (
            ARRAY[-1] as trace_address,
            project_contract_address,
            trades.order_uid,
-           trader,
+           -- ETH Flow orders have trader = sender of orderCreation.
+           case when sender is not null then sender else trader end as trader,
```
Comment (bh2smith) [inline dbt_subprojects/dex/models/_projects/cow_protocol/gnosis/cow_protocol_gnosis_trades.sql:190]: Nice find!

File: dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/cow_protocol_batches.sql [hunk 3]
```diff
@@ -51,26 +51,4 @@ FROM
...
-            solver_address,
-            tx_hash,
-            gas_price,
-            gas_used,
-            tx_cost_usd,
-            fee_value,
-            call_data_size,
-            unwraps,
-            token_approvals
-        FROM {{ ref('cow_protocol_arbitrum_batches') }}
```
Comment (bh2smith) [inline dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/cow_protocol_batches.sql:75]: Why remove this union?

**Review by bh2smith — APPROVED** [review]
Seems legit.

File: dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/cow_protocol_batches.sql (hunk 3 above)
Comment (fleupold) [reply dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/cow_protocol_batches.sql:75]: Great catch (I wanted to remove some base references I had locally, but went too far)

File: dbt_subprojects/dex/models/_projects/cow_protocol/arbitrum/cow_protocol_arbitrum_trades.sql (hunk 1 above)
Comment (fleupold) [reply dbt_subprojects/dex/models/_projects/cow_protocol/arbitrum/cow_protocol_arbitrum_trades.sql:14]: Will do this for the ones I actually contribute to

File: dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/cow_protocol_batches.sql (hunk 3 above)
Comment (jeff-dude) [reply dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/cow_protocol_batches.sql]: hitting home for me on this one, i'm a reds fan!

**Review by jeff-dude — APPROVED** [review]
this works for me, happy to include in merge here shortly if all good 🙏

**Pull request merged**